lib/preflight.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const { typeErrors: E } = require('./util/constants');
+const { isNonArrayObject } = require('./util');
+const { PreflightTest } = require('./preflighttest');
+
+
+/**
+ * Test the connection to Twilio servers, and generate report for the connection.
+ *   This function uses the given tokens to establish a room connection
+ *   and generates a report on success. it can be used to verify end to end
+ *   connectivity,
+ * @alias module:twilio-video.testPreflight
+ * @param {string} publisherToken - The Access Token string for the participant. This participant will publish tracks.
+ * @param {string} subscriberToken - The Access Token string for the participant. This participant will subscribe to tracks.
+ * @param {PreflightOptions} [options] - Options to override the default behavior
+ * @returns {PreflightTest} - a {@link PreflightTest} object that can be used to monitor the progress of the test
+ * @example
+ * var Video = require('twilio-video');
+ * var publisherToken = getAccessToken('alice');
+ * var subscriberToken = getAccessToken('bob');
+ * var preflightTest = Video.testPreflight(publisherToken, subscriberToken);
+ * preflightTest.on('completed', function(report) {
+ *   console.log("Test completed in " + report.testTiming.duration + " milliseconds.");
+ *   console.log(" It took " + report.networkTiming.connect.duration + " milliseconds to connect");
+ *   console.log(" It took " + report.networkTiming.media.duration + " milliseconds to receive media");
+ *   console.log(" Your network score was: " + report.stats.networkQuality); // only available for group rooms.
```

Comment:
How would a developer choose what room type to use? Does the preflight test use the room name that is contained in the token? Or does this test create a new room using the default room type from the Twilio console. I'm wondering if it would be useful if the docs went into a little more detail about this.

Reply:
Good point, Yes I will add more details in the js docs about this. 
The test uses the token supplied, so it will  use  the room mentioned in the token, if it does not exist it will be created using the default room type from console.

I think developers typically would not want to specify room-type, because there app either uses group room or p2p room - whichever room type the app uses should be set as default room type in console.

But if they do want to call it with different room-type they should create room before calling the api - as our integration test does.